<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>AdvancedTestHElib</class>
 <widget class="QWidget" name="AdvancedTestHElib">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1064</width>
    <height>1011</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>HElib 的高级测试</string>
  </property>
  <property name="styleSheet">
   <string notr="true">QStackedWidget,  QLabel,QPushButton, QRadioButton, QCheckBox, 
QGroupBox, QStatusBar, QToolButton, QComboBox, QDialog {
    background-color: #222222;
    color: #BBBBBB;
    font-family: &quot;Segoe UI&quot;;
}
QLabel {
    border: none;
    padding: 0;
    background: none;
}

QPushButton {
    border: 1px solid #333333;
    padding: 4px;
    min-width: 65px;
    min-height: 12px;
}

QPushButton:hover {
    background-color: #333333;
    border-color: #444444;
}

QPushButton:pressed {
    background-color: #111111;
    border-color: #333333;
    color: yellow;
}

QPushButton:disabled {
    color: #333333;
}

QComboBox {
    background-color: #AAA;
    border: 1px solid #555;
    color: black;
}

QComboBox::drop-down {
    subcontrol-origin: padding;
    subcontrol-position: top right;
    border-left: 1px solid #333333;
}
QAbstractScrollArea, QLineEdit, QTextEdit, QAbstractSpinBox, QComboBox {
    border-color: #333333;
    border: 1px solid #333333;

}</string>
  </property>
  <widget class="QLabel" name="label">
   <property name="geometry">
    <rect>
     <x>40</x>
     <y>20</y>
     <width>241</width>
     <height>81</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Segoe UI</family>
     <pointsize>18</pointsize>
    </font>
   </property>
   <property name="mouseTracking">
    <bool>false</bool>
   </property>
   <property name="styleSheet">
    <string notr="true">QLabel {
    border: none;
    padding: 0;
    background: none;
}</string>
   </property>
   <property name="text">
    <string>HElib 的高级测试</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_2">
   <property name="geometry">
    <rect>
     <x>40</x>
     <y>120</y>
     <width>91</width>
     <height>41</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Segoe UI</family>
     <pointsize>16</pointsize>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">QLabel {
    border: none;
    padding: 0;
    background: none;
}</string>
   </property>
   <property name="text">
    <string>设置参数</string>
   </property>
  </widget>
  <widget class="QPushButton" name="pushButton">
   <property name="geometry">
    <rect>
     <x>180</x>
     <y>120</y>
     <width>91</width>
     <height>41</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Segoe UI</family>
     <pointsize>12</pointsize>
    </font>
   </property>
   <property name="text">
    <string>参数生成器</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_4">
   <property name="geometry">
    <rect>
     <x>40</x>
     <y>200</y>
     <width>41</width>
     <height>41</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Segoe UI</family>
     <pointsize>14</pointsize>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">QLabel {
    border: none;
    padding: 0;
    background: none;
}</string>
   </property>
   <property name="text">
    <string>p =</string>
   </property>
  </widget>
  <widget class="QLineEdit" name="pri">
   <property name="geometry">
    <rect>
     <x>80</x>
     <y>210</y>
     <width>113</width>
     <height>25</height>
    </rect>
   </property>
   <property name="text">
    <string>4999（默认）</string>
   </property>
  </widget>
  <widget class="QLineEdit" name="SetM">
   <property name="geometry">
    <rect>
     <x>270</x>
     <y>210</y>
     <width>113</width>
     <height>25</height>
    </rect>
   </property>
   <property name="text">
    <string>32109（默认）</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_5">
   <property name="geometry">
    <rect>
     <x>220</x>
     <y>200</y>
     <width>41</width>
     <height>41</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Segoe UI</family>
     <pointsize>14</pointsize>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">QLabel {
    border: none;
    padding: 0;
    background: none;
}</string>
   </property>
   <property name="text">
    <string>m =</string>
   </property>
  </widget>
  <widget class="QLineEdit" name="SetR">
   <property name="geometry">
    <rect>
     <x>460</x>
     <y>210</y>
     <width>113</width>
     <height>25</height>
    </rect>
   </property>
   <property name="text">
    <string>1（默认）</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_6">
   <property name="geometry">
    <rect>
     <x>420</x>
     <y>200</y>
     <width>41</width>
     <height>41</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Segoe UI</family>
     <pointsize>14</pointsize>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">QLabel {
    border: none;
    padding: 0;
    background: none;
}</string>
   </property>
   <property name="text">
    <string>r =</string>
   </property>
  </widget>
  <widget class="QLineEdit" name="SetBits">
   <property name="geometry">
    <rect>
     <x>80</x>
     <y>270</y>
     <width>113</width>
     <height>25</height>
    </rect>
   </property>
   <property name="text">
    <string>300（默认）</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_7">
   <property name="geometry">
    <rect>
     <x>20</x>
     <y>260</y>
     <width>51</width>
     <height>41</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Segoe UI</family>
     <pointsize>14</pointsize>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">QLabel {
    border: none;
    padding: 0;
    background: none;
}</string>
   </property>
   <property name="text">
    <string>bits =</string>
   </property>
  </widget>
  <widget class="QLineEdit" name="SetC">
   <property name="geometry">
    <rect>
     <x>270</x>
     <y>270</y>
     <width>113</width>
     <height>25</height>
    </rect>
   </property>
   <property name="text">
    <string>2（默认）</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_8">
   <property name="geometry">
    <rect>
     <x>230</x>
     <y>260</y>
     <width>41</width>
     <height>41</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Segoe UI</family>
     <pointsize>14</pointsize>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">QLabel {
    border: none;
    padding: 0;
    background: none;
}</string>
   </property>
   <property name="text">
    <string>c =</string>
   </property>
  </widget>
  <widget class="QLineEdit" name="test_number">
   <property name="geometry">
    <rect>
     <x>460</x>
     <y>270</y>
     <width>113</width>
     <height>25</height>
    </rect>
   </property>
   <property name="text">
    <string>3</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_9">
   <property name="geometry">
    <rect>
     <x>390</x>
     <y>260</y>
     <width>61</width>
     <height>41</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Segoe UI</family>
     <pointsize>11</pointsize>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">QLabel {
    border: none;
    padding: 0;
    background: none;
}</string>
   </property>
   <property name="text">
    <string>测试次数</string>
   </property>
  </widget>
  <widget class="QLineEdit" name="plain_size">
   <property name="geometry">
    <rect>
     <x>700</x>
     <y>210</y>
     <width>113</width>
     <height>25</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>12</pointsize>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">QAbstractScrollArea, QLineEdit, QTextEdit, QAbstractSpinBox, QComboBox {
    border-color: #333333;
    border: 1px solid #333333;

}</string>
   </property>
   <property name="text">
    <string>2</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_10">
   <property name="geometry">
    <rect>
     <x>580</x>
     <y>200</y>
     <width>121</width>
     <height>41</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Segoe UI</family>
     <pointsize>12</pointsize>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">QLabel {
    border: none;
    padding: 0;
    background: none;
}</string>
   </property>
   <property name="text">
    <string>明文大小不超过</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_11">
   <property name="geometry">
    <rect>
     <x>610</x>
     <y>260</y>
     <width>71</width>
     <height>41</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Segoe UI</family>
     <pointsize>12</pointsize>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">QLabel {
    border: none;
    padding: 0;
    background: none;
}</string>
   </property>
   <property name="text">
    <string>测试类型</string>
   </property>
  </widget>
  <widget class="QComboBox" name="TestType">
   <property name="geometry">
    <rect>
     <x>700</x>
     <y>270</y>
     <width>111</width>
     <height>25</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true">QComboBox {
    background-color: #AAA;
    border: 1px solid #555;
    color: black;
}

QComboBox::drop-down {
    subcontrol-origin: padding;
    subcontrol-position: top right;
    border-left: 1px solid #333333;
}</string>
   </property>
   <item>
    <property name="text">
     <string>Add测试</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>Add Plain测试</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>Mult测试</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>Mult Plain测试</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>Square测试</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>Negation测试</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>Sub测试</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>Xor测试</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>nXor测试</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>Rotate Random测试</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>Encryption测试</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>Decryption测试</string>
    </property>
   </item>
  </widget>
  <widget class="QtCharts::QChartView" name="graphicsView_2">
   <property name="geometry">
    <rect>
     <x>570</x>
     <y>760</y>
     <width>451</width>
     <height>211</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true">QAbstractScrollArea, QLineEdit, QTextEdit, QAbstractSpinBox, QComboBox {
    border-color: #333333;
    border: 1px solid #333333;

}</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_12">
   <property name="geometry">
    <rect>
     <x>740</x>
     <y>300</y>
     <width>121</width>
     <height>41</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Segoe UI</family>
     <pointsize>13</pointsize>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">QLabel {
    border: none;
    padding: 0;
    background: none;
}</string>
   </property>
   <property name="text">
    <string>运算解密后明文</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_13">
   <property name="geometry">
    <rect>
     <x>220</x>
     <y>520</y>
     <width>71</width>
     <height>31</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Segoe UI</family>
     <pointsize>13</pointsize>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">QLabel {
    border: none;
    padding: 0;
    background: none;
}</string>
   </property>
   <property name="text">
    <string>测试结果</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_14">
   <property name="geometry">
    <rect>
     <x>760</x>
     <y>520</y>
     <width>71</width>
     <height>31</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Segoe UI</family>
     <pointsize>13</pointsize>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">QLabel {
    border: none;
    padding: 0;
    background: none;
}</string>
   </property>
   <property name="text">
    <string>图像表示</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_15">
   <property name="geometry">
    <rect>
     <x>210</x>
     <y>300</y>
     <width>91</width>
     <height>41</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Segoe UI</family>
     <pointsize>13</pointsize>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">QLabel {
    border: none;
    padding: 0;
    background: none;
}</string>
   </property>
   <property name="text">
    <string>加密前明文</string>
   </property>
  </widget>
  <widget class="QTextEdit" name="plain">
   <property name="geometry">
    <rect>
     <x>30</x>
     <y>350</y>
     <width>451</width>
     <height>161</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true">QAbstractScrollArea, QLineEdit, QTextEdit, QAbstractSpinBox, QComboBox {
    border-color: #424242;
    border: 1px solid #424242;

}</string>
   </property>
  </widget>
  <widget class="QtCharts::QChartView" name="graphicsView">
   <property name="geometry">
    <rect>
     <x>570</x>
     <y>570</y>
     <width>451</width>
     <height>191</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true">QAbstractScrollArea, QLineEdit, QTextEdit, QAbstractSpinBox, QComboBox {
    border-color: #333333;
    border: 1px solid #333333;

}</string>
   </property>
  </widget>
  <widget class="QTextEdit" name="result">
   <property name="geometry">
    <rect>
     <x>30</x>
     <y>570</y>
     <width>451</width>
     <height>401</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true">QAbstractScrollArea, QLineEdit, QTextEdit, QAbstractSpinBox, QComboBox {
    border-color: #333333;
    border: 1px solid #333333;

}</string>
   </property>
  </widget>
  <widget class="QTextEdit" name="plain_end">
   <property name="geometry">
    <rect>
     <x>570</x>
     <y>350</y>
     <width>451</width>
     <height>161</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true">QAbstractScrollArea, QLineEdit, QTextEdit, QAbstractSpinBox, QComboBox {
    border-color: #333333;
    border: 1px solid #333333;

}</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_16">
   <property name="geometry">
    <rect>
     <x>720</x>
     <y>950</y>
     <width>61</width>
     <height>17</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true">QLabel {
    border: none;
    padding: 0;
    background: none;
}</string>
   </property>
   <property name="text">
    <string>密文时间</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_17">
   <property name="geometry">
    <rect>
     <x>820</x>
     <y>950</y>
     <width>16</width>
     <height>17</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true">border-image: url(:/choose/image/绿色.png);</string>
   </property>
   <property name="text">
    <string/>
   </property>
  </widget>
  <widget class="QLabel" name="label_18">
   <property name="geometry">
    <rect>
     <x>850</x>
     <y>950</y>
     <width>61</width>
     <height>17</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true">QLabel {
    border: none;
    padding: 0;
    background: none;
}</string>
   </property>
   <property name="text">
    <string>明文时间</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_19">
   <property name="geometry">
    <rect>
     <x>690</x>
     <y>950</y>
     <width>16</width>
     <height>17</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true">border-image: url(:/choose/image/蓝色.png);</string>
   </property>
   <property name="text">
    <string/>
   </property>
  </widget>
  <widget class="QPushButton" name="start">
   <property name="geometry">
    <rect>
     <x>710</x>
     <y>130</y>
     <width>89</width>
     <height>41</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Segoe UI</family>
     <pointsize>12</pointsize>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">QPushButton {
    border: 1px solid #333333;
    padding: 4px;
    min-width: 65px;
    min-height: 12px;
}

QPushButton:hover {
    background-color: #333333;
    border-color: #444444;
}

QPushButton:pressed {
    background-color: #111111;
    border-color: #333333;
    color: yellow;
}

QPushButton:disabled {
    color: #333333;
}</string>
   </property>
   <property name="text">
    <string>开始测试</string>
   </property>
  </widget>
  <widget class="QPushButton" name="return_2">
   <property name="geometry">
    <rect>
     <x>870</x>
     <y>130</y>
     <width>89</width>
     <height>41</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Segoe UI</family>
     <pointsize>12</pointsize>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">QPushButton {
    border: 1px solid #333333;
    padding: 4px;
    min-width: 65px;
    min-height: 12px;
}

QPushButton:hover {
    background-color: #333333;
    border-color: #444444;
}

QPushButton:pressed {
    background-color: #111111;
    border-color: #333333;
    color: yellow;
}

QPushButton:disabled {
    color: #333333;
}</string>
   </property>
   <property name="text">
    <string>返回</string>
   </property>
  </widget>
  <widget class="QCheckBox" name="checkBox">
   <property name="geometry">
    <rect>
     <x>830</x>
     <y>210</y>
     <width>151</width>
     <height>23</height>
    </rect>
   </property>
   <property name="text">
    <string>是否测试电路深度</string>
   </property>
  </widget>
  <widget class="QCheckBox" name="checkBox_2">
   <property name="geometry">
    <rect>
     <x>830</x>
     <y>270</y>
     <width>171</width>
     <height>23</height>
    </rect>
   </property>
   <property name="text">
    <string>是否开启密钥交换矩阵</string>
   </property>
  </widget>
  <widget class="QLabel" name="testing">
   <property name="geometry">
    <rect>
     <x>420</x>
     <y>110</y>
     <width>161</width>
     <height>61</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Segoe UI</family>
     <pointsize>15</pointsize>
    </font>
   </property>
   <property name="text">
    <string>测试中，请稍等...</string>
   </property>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QtCharts::QChartView</class>
   <extends>QGraphicsView</extends>
   <header location="global">qchartview.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
